src/build_scripts/buildlib.kokkos

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+
+from standard_script_setup import *
+from CIME.utils import expect, run_bld_cmd_ensure_logging, run_cmd_no_fail
+from CIME.case import Case
+
+logger = logging.getLogger(__name__)
+
+def parse_command_line(args, description):
+###############################################################################
+    parser = argparse.ArgumentParser(
+        usage="""\n{0} [--debug]
+OR
+{0} --verbose
+OR
+{0} --help
+
+\033[1mEXAMPLES:\033[0m
+    \033[1;32m# Run \033[0m
+    > {0}
+""" .format (os.path.basename(args[0])),
+
+description=description,
+
+formatter_class=argparse.ArgumentDefaultsHelpFormatter
+)
+
+    CIME.utils.setup_standard_logging_options(parser)
+
+    parser.add_argument("buildroot",
+                        help="build path root")
+
+    parser.add_argument("installpath",
+                        help="install path ")
+
+    parser.add_argument("caseroot", nargs="?", default=os.getcwd(),
+                        help="Case directory to build")
+
+    args = CIME.utils.parse_args_and_handle_standard_logging_options(args, parser)
+
+    return args.buildroot, args.installpath, args.caseroot
+
+###############################################################################
+def buildlib(bldroot, installpath, case):
+###############################################################################
+    srcroot = case.get_value("SRCROOT")
+    kokkos_dir = os.path.join(srcroot, "externals", "kokkos")
+    expect(os.path.isdir(kokkos_dir), "Missing kokkos submodule")
+
+    # We want to get the compiler and kokkos_options from Macros.make
+    # (generated from config_compilers.xml), but we want to otherwise
+    # let kokkos control flags
+    kokkos_options = run_cmd_no_fail("make -f Macros.make -p | grep KOKKOS_OPTIONS").split(":=")[-1].strip()
+    cxx            = run_cmd_no_fail("make -f Macros.make -p | grep SCXX").split(":=")[-1].strip()
+
+    gmake_cmd = case.get_value("GMAKE")
+    gmake_j = case.get_value("GMAKE_J")
+
```

Comment:
If you get the Makefile options and pass them to gmake here you can avoid both environment variables and xmlquery calls.